### PR TITLE
builtins stubtest exceptions

### PR DESCRIPTION
**Pull request by hatal175**
- I've just left 'class function' - anyone can explain what it is and why it is there?

### CODE REVIEW

**Comment by hauntsaninja** [comment]
There's https://github.com/python/typeshed/issues/2999 for builtins.function

File: tests/stubtest_whitelists/py3_common.txt [hunk 1]
```diff
@@ -49,30 +49,15 @@ asyncio.locks.Condition.locked
...
-builtins.bytearray.append
-builtins.bytearray.extend
-builtins.bytearray.maketrans
-builtins.bytes.__float__
-builtins.bytes.__int__
-builtins.bytes.maketrans
-builtins.classmethod.__get__
-builtins.ellipsis
-builtins.filter  # not a function at runtime
+builtins.classmethod.__get__  # this function can accpet no value for the type parameter.
```
Comment (JelleZijlstra) [inline tests/stubtest_whitelists/py3_common.txt:52]:
```suggestion
builtins.classmethod.__get__  # this function can accept no value for the type parameter.
```

File: tests/stubtest_whitelists/py3_common.txt [hunk 2]
```diff
@@ -49,30 +49,15 @@ asyncio.locks.Condition.locked
...
-builtins.property.__get__
-builtins.property.fdel
-builtins.property.fget
-builtins.property.fset
-builtins.reversed  # not a function at runtime
-builtins.staticmethod.__get__
-builtins.zip  # not a function at runtime
+builtins.object.__init__  # default C signature is incorrect
+builtins.property.__get__  # this function can accpet no value for the type parameter.
+builtins.staticmethod.__get__  # this function can accpet no value for the type parameter.
```
Comment (JelleZijlstra) [inline tests/stubtest_whitelists/py3_common.txt:60]:
```suggestion
builtins.property.__get__  # this function can accept no value for the type parameter.
builtins.staticmethod.__get__  # this function can accept no value for the type parameter.
```

**Comment by JelleZijlstra** [comment]
The `bidict`failure is a bit nasty: it's on a line `key = next((reversed if last else iter)(self))`. This doesn't work any more because now mypy infers `object` for `reversed if last else iter`. Still, this sort of issue seems rare so I'm OK with moving forward with these changes.

File: stdlib/builtins.pyi [hunk 3]
```diff
@@ -472,14 +472,12 @@ class bytes(ByteString):
...
     def fromhex(cls, __s: str) -> bytes: ...
-    @classmethod
-    def maketrans(cls, frm: bytes, to: bytes) -> bytes: ...
+    @staticmethod
+    def maketrans(__frm: bytes, __to: bytes) -> bytes: ...
     def __len__(self) -> int: ...
     def __iter__(self) -> Iterator[int]: ...
     def __str__(self) -> str: ...
     def __repr__(self) -> str: ...
-    def __int__(self) -> int: ...
```
Comment (JelleZijlstra) [inline stdlib/builtins.pyi:481]: I wonder if these were added on purpose to make bytes fulfill the `SupportsInt` and `SupportsFloat` protocols. `int(b'12')` works at runtime. Probably safer to keep them.

Comment (hatal175) [reply stdlib/builtins.pyi:481]: 1. no mypy primer diffs
2. the int construct specifically gets bytes (so does float) and treats it as a string
3. Other functions that receive SupportsInt fail on bytes - itertools.count(b'1',1)

Comment (JelleZijlstra) [reply stdlib/builtins.pyi:481]: Sounds good, I will merge

**Comment by ktbarrett** [comment]
Why were all these overloaded functions changed to classes? This breaks deduction in [certain contexts](https://mypy-play.net/?mypy=latest&python=3.10&gist=6552f35c9f665482d43d8786af3fed26).

**Comment by Akuli** [comment]
I'm not yet convinced that this PR has anything to do with that. I added this function to your mypy playground link, based on what this PR deletes, and used it instead of `zip`:

```
def myzip(__iter1: Iterable[_T1], __iter2: Iterable[_T2]) -> Iterator[Tuple[_T1, _T2]]:
    return zip(__iter1, __iter2)
```

It still gives the same number of errors.

**Pull request merged**